solutions/0400-0499/0445-Add-Two-Numbers-II.md

```diff
@@ -0,0 +1,145 @@
+---
+description: >-
+  Author: @Kavita613 |
+  https://leetcode.com/problems/add-two-numbers-ii/
+tags: [Linked List, Math, Stack]
+---
+
+# 445 - Add two Numbers II (Medium) 
+
+## Problem Link
+
+https://leetcode.com/problems/add-two-numbers-ii/
+
+## Problem Statement
+
+You are given two non-empty linked lists representing two non-negative integers. The most significant digit comes first and each of their nodes contains a single digit. 
+Add the two numbers and return the sum as a linked list. 
+You may assume the two numbers do not contain any leading zero, except the number 0 itself.
+
+**Example 1:**
+
+<img src="https://assets.leetcode.com/uploads/2021/04/09/sumii-linked-list.jpg" alt="Photo" height="160" />
+
+```
+Input: l1 = [7,2,4,3], l2 = [5,6,4]
+Output: [7,8,0,7] 
+```
+
+**Example 2:**
+
+```
+Input: l1 = [2,4,3], l2 = [5,6,4]
+Output: [8,0,7]
+```
+
+**Example 3:**
+
+```
+Input: l1 = [0], l2 = [0]
+Output: [0]
+```
+
+**Constraints:**
+
+- The number of nodes in each linked list is in the range `[1, 100]`.
+- `0 <= Node.val <= 9`
+- It is guaranteed that the list represents a number that does not have leading zeros.
+
+**Follow up:**  
+Could you solve it without reversing the input lists?
+
+## Approach 1: Recursion
+First, we will add zeros to the start of the smaller linked list such that both the lists become of equal size and then we will use recursion to perform 
+addition start to end.  
+Now, we will use recursion(like reverse traversal) to the end of both the lists and start addition from the end. After each recursion ends, $l1$ and $l2$ will be 
+waiting at the previous nodes. Here we will pass $carry$ from current recursive function to the previous recursive function, for which we can pass the reference 
+variable $carry$ to function so that the changes made to $carry$ reflect through all the recursive calls made.
+
+
+<Tabs>
+
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Kavita613"/>
+
+```cpp
+/**
```

Comment:
1. `addDigits` should be placed inside
2. need more comments on your code. beginners won't understand that
3. your explanation is not detailed enough. please write more and add some images to help if possible.